python/cudf/cudf/testing/csv.py

```diff
@@ -0,0 +1,184 @@
+import copy
+import json
+import logging
+import os
+import random
+import sys
+
+import cudf
+from cudf.testing.utils import _generate_rand_meta
+from cudf.tests import dataset_generator as dg
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+class CSVReader(object):
+    def __init__(
+        self,
+        file_name="temp_csv.csv",
+        dirs=None,
+        max_rows=4096,
+        max_columns=1000,
+    ):
+        self._inputs = []
+        self._file_name = file_name
+        self._max_rows = max_rows
+        self._max_columns = max_columns
+
+        for i, path in enumerate(dirs):
+            if i == 0 and not os.path.exists(path):
+                raise FileNotFoundError(f"No {path} exists")
+
+            if os.path.isfile(path) and path.endswith("_crash.json"):
+                self._load_params(path)
+            else:
+                for i in os.listdir(path):
+                    file_name = os.path.join(path, i)
+                    if os.path.isfile(file_name) and file_name.endswith(
+                        "_crash.json"
+                    ):
+                        self._load_params(file_name)
+        self._regression = True if self._inputs else False
+        self._idx = 0
+        self._current_params = {}
+
+    def _load_params(self, path):
+        with open(path, "r") as f:
+            params = json.load(f)
+        self._inputs.append(params)
```

Comment:
This method seems shared across all of the Reader / Writer classes, maybe build a parent class to abstract it?

Reply:
Done